javascript/javascript1/week4/voice-assist.js

```diff
@@ -0,0 +1,120 @@
+// list all accepted comands in object
+const acceptedCommands = {
+    logMyName: "what is my name",
+    logToDo: "what is on my todo",
+    logDate: "what day is it today",
+    resetUser: "reset user", // adding one more command
+    hello: "hello my name is ",
+    addToDo: "add ",
+    removeToDo: "remove ",
+    calc: "what is ",
+    timer: "set a timer for "
+}
+// global variables to hold user name, and todos
+let myName = "";
+let toDos = [];
+
+function getReply(command) {
+    // make command case insensetive, and exclude excess spaces and ? signs in the end
+    let com = command.toLowerCase().trim();
+    com = com.replace(/\?*$/, '');
+
+    // first deal with full command string found in accepted commands
+    if (Object.values(acceptedCommands).includes(com)) {
+        switch (com) {
+            case acceptedCommands.logMyName:
+                return myName ? `Your name is ${myName}` : "You did not introduce yourself yet!";
+            case acceptedCommands.logToDo:
+                return toDos.length ? `Here's list of your ToDos:\n - ${toDos.join("\n - ")}` : "You have nothing on the list";
+            case acceptedCommands.logDate:
+                return new Date().toDateString();
+            case acceptedCommands.resetUser:
+                myName = "";
+                toDos = [];
+                return "Hello what is your name?"
+        }
+        // next deal with name saving
+    } else if (com.startsWith(acceptedCommands.hello)) {
+        // checking if name already given
+        myName = myName ?
+            `You already said your name is ${myName}`
+            //works for name with few words with spaces, first I substring and capitalize Name again
+            : com.substring(acceptedCommands.hello.length).split(" ").map(w => w.charAt(0).toUpperCase() + w.slice(1)).join(" ");
+        return `Nice to meet you ${myName}`
+
+        // deal with toDos
+        // parse comand string assuming all comands will end with "to/from my todo"
+    } else if (com.startsWith(acceptedCommands.addToDo)) {
+        const todo = com.replace("add ", "").replace(" to my todo", "");
+        toDos.push(todo)
+        return `Ok. ${todo} added to your todo`;
+    } else if (com.startsWith(acceptedCommands.removeToDo)) {
+        const todo = com.replace("remove ", "").replace(" from my todo", "");
+        const removed = toDos.filter(td => td === todo);
+        return removed.length ? `Done. ${removed[0]} was removed from todo` : `Didn't find ${todo} in your todo.`
+
+        // calc and timer moved to another function for better readability
+    } else if (com.startsWith(acceptedCommands.calc)) { // there other commands that start with "What is..." but they were dealed with before so only calc left
+        return getCalcResult(com.replace(acceptedCommands.calc, ""));
+    } else if (com.startsWith(acceptedCommands.timer)) {
+        return startTimer(com.replace(acceptedCommands.timer, ""))
+    } else {
```

Comment:
Redundant else-statement.

However it is nice to know that this is the final situation, so eventhough it is logically unnecessary, it might be good for the semantics(conveying the meaning of what the function does) - and that is also important.